gatherers/manifest.js

```diff
@@ -54,34 +32,62 @@ class Manifest extends Gather {
     };
   }
 
+  /**
+   * Creates a listener for manifests in network requests, a load timeout, and
+   * a promise they resolve when complete.
+   * @private
+   */
+  static _resolveOnManifest() {
```

Comment:
it's not much more than what it's replacing :)

I do agree that some of this could easily live in the driver, though. If any gather can listen for particular requests, and any gather can extend the time that network recording is going on, this can easily be interleaved with `load.js` (the default trace/network recording gather) if we go with some of @paulirish's ideas in #81